src/basics.md

```diff
@@ -276,6 +277,26 @@ fn main() {
 }
 ```
 
+[ex-rand-passwd]: #ex-rand-passwd
+<a name="ex-rand-passwd"></a>
+## Create random passwords from a set of allowed characters
+
+[![rand-badge]][rand] [![cat-text-processing-badge]][cat-text-processing]
+
+Randomly generates a string of given length ASCII characters in the range `A-Z, a-z, 0-9`.
```

Comment:
Can you give more details, such as [`gen_ascii_chars`](https://docs.rs/rand/0.3.18/rand/trait.Rng.html#method.gen_ascii_chars) is the method which a hardcoded range of character.

Plus, I would show the [`choose`](https://docs.rs/rand/0.3.18/rand/trait.Rng.html#method.choose) method which is a user defined set of allowed characters. Maybe wait for the green flag of @budziq for this one.

Reply:
Agreed on both arguments:
- lets add links to key identifiers in the description.
- I'd suggest to add two snippets here the second one being an iterator based in `choose` like the one used to implement `gen_ascii_chars` but taking arbitrary bytestring in the constructor, then used in identical fashion as the first trivial snippet. That would be very cool!